llvm-bazel/llvm-project-overlay/mlir/tblgen.bzl

```diff
@@ -4,7 +4,128 @@
 
 """BUILD extensions for MLIR table generation."""
 
-def gentbl(name, tblgen, td_file, tbl_outs, td_srcs = [], td_includes = [], td_relative_includes = [], strip_include_prefix = None, test = False):
+# A provider with one field, transitive_sources.
+TdFiles = provider(fields = ["transitive_sources", "transitive_includes"])
+
+def _get_dep_transitive_srcs(dep):
+    if TdFiles in dep:
+        return dep[TdFiles].transitive_sources
+    return dep[DefaultInfo].files
```

Comment:
Yeah that was the intent and I went back and forth on whether that was a good idea. When I was moving over all the rules in this file it made things a bit easier because I had an intermediate state that worked. I think it might also ease the rollout because someone won't have to switch over all their dependencies in order to adopt the new format.

What about leaving filegroups allowed here for now with a TODO to remove them once a reasonable transition period has passed?

Reply:
Your call, TODO sgtm.